### ISSUE TITLE
Reorganize .R files around endpoints?

### ISSUE THREAD

**Issue by jmobrien** [ropensci/qualtRics#303]
I've been thinking about the need to revamp URL generation for #294, the number of new and pending new functions, and the casual discussion about where to place new things such as in #299 and/or #302.  Also was doing some things over at *haven* that had me looking at a recent [organization change](https://github.com/tidyverse/haven/commit/b093a6bd20dfee4cf09c3f70a99c8bf80e3803ea).  

Got me wondering if we might do well to start a file structure that organizes functions differently.  Currently wondering about organizing files around major API endpoints, like the ones for [surveys](https://api.qualtrics.com/60d24f6897737-qualtrics-survey-api), [responses](https://api.qualtrics.com/206a07d54ca31-surveys-response-import-export-api), and [mailing lists](https://api.qualtrics.com/0b9b0d24a7a18-mailing-lists).  

Related *potential* ideas, pretty independent so could consider separately:
- organize files by major endpoint groups
- files also have endpoint-associated info/functions that guide url building (since this might help with swapping out to mock servers for tests, and/or can be re-used to build new functions.)
- check functions unique to specific endpoints/functions go in section in same file for easier review/management
- general-purpose check functions folded into utils.R to reduce clutter (delete checks.R)
- others?

**Comment by juliasilge** [comment]
Oh yeah, I think this package could do with some refactoring and reorganization. I personally find it hard to locate something at times! I think `utils.R` has _way_ too much in it, and we should have a file for request building, a file for result parsing, etc.

Like you said, not all of this has to be done at once; some gradual improvements would really get us a better maintenance experience.

**Comment by jmobrien** [comment]
Yeah, I cut down `utils.R` a lot in the last reorganize, and it's still got a lot.  Okay.  I'll poke at it as I finish up some of this other stuff.
